jablib/src/test/java/org/jabref/logic/citation/SearchCitationsRelationsServiceTest.java

```diff
@@ -21,7 +23,7 @@ class SearchCitationsRelationsServiceTest {
     /**
      * Creates a mock CitationFetcher that returns specific results for citations and references
      */
-    private CitationFetcher createMockFetcher(BibEntry targetEntry, List<BibEntry> citationsToReturn, List<BibEntry> referencesToReturn) {
+    private CitationFetcher createMockFetcher(BibEntry targetEntry, List<BibEntry> citationsToReturn, List<BibEntry> referencesToReturn, Optional<Integer> citationCount) {
```

Comment:
`citationCount` should be `@NonNull` - mixing Optionals and `null` (line 42) is a bad practise. -- `Optionals` remove the need to `null` checking.

Even in tests